webpack/assets/javascripts/react_app/redux/API/APIRequest.js

```diff
@@ -1,24 +1,45 @@
 import { API } from './';
+import { actionTypeGenerator } from './APIActionTypeGenerator';
+import {
+  formatSuccess as defaultFormatSuccess,
+  formatFailure as defaultFormatFailure,
+  formatPending as defaultFormatPending,
+} from './APIHelpers';
 
-export const get = async (payload, url, store, actionTypes) => {
-  store.dispatch({
-    type: actionTypes.REQUEST,
+export const get = async (action, store) => {
+  const {
+    key,
+    url,
     payload,
+    actionTypes,
+    formatPending = defaultFormatPending,
+    formatSuccess = defaultFormatSuccess,
+    formatFailure = defaultFormatFailure,
+  } = action;
+  const { REQUEST, SUCCESS, FAILURE } = actionTypeGenerator(key, actionTypes);
+
+  store.dispatch({
+    key,
+    type: REQUEST,
+    payload: formatPending({ payload }),
   });
+
   try {
     const { data } = await API.get(
       url,
       payload.headers || {},
       payload.params || {}
     );
     store.dispatch({
-      type: actionTypes.SUCCESS,
-      payload: { ...payload, ...data },
+      key,
+      type: SUCCESS,
+      payload: formatSuccess({ data, payload }),
```

Comment:
I see it odd that the `payload` contains the `data` and the `payload`.
I see `payload` as what we send to the server and `data` as what we received from the server and I believe they should be seperated.

Reply:
although today they are already mixed inside the payload as: 
`payload: { ...payload, ...data }` - how would you separate them in the store? can you give an example? 

while adding the formatter, I let the consumer decide how they want to manage their store by giving them the `data` and `payload`, and the default behavior will remain as it is now.